### PR TITLE
Reorganize topographical View

### PR DESCRIPTION
**Pull request by haesleinhuepf**
Hi Marcelo @zoccoler,

this is a refactoring of your Topographical views code. The results should be identical with your implementation. I changed the internal code because: The generated image/points/surface were strongly related to the layer properties. Thus, I moved those code sections together. The function that produces the surface now also produces the surface layer properties.
This also allows to simplify the widget code so that there are not if-blocks anymore because I'm using an Enum ([related discussion](https://forum.image.sc/t/empty-dropdown-in-napari-after-trial-of-widget-createion/67086)) to allow the user to choose what to visualize. The user-interface now looks like this:

image

If you think that's a bad idea, I can revert it to the checkboxes, but my intention was: Users may have one preferred way of visualizing the profile and will likely not view multiple views (image, points, surfaces) at a time. Thus, a pull-down may be easier to configure than three checkboxes. I think it was like that before, no? The screen recording looks like...

I simplified the test code as it was hard to read and understand. We should not write tests that take a list of functions as input and then test them. Write one test for one function. In case a test breaks, the bug is then much easier to find. See also: https://youtu.be/gSBR34m1OCE?t=539 at 9:00

I also renamed some variable to become English words and for clarity. I also renamed "sample_factor" to "step_size" so that the term is everywhere the same. Furthermore, I believe the sample_factor was actually a sample_divisor.

The tests pass and I tested the four functions from the user interface (Menu Plugins > napari-plot-profile > Topographical ... and Tools > Visualization > Topographical View) manually. 

Would you mind taking another look if everything is ok from your perspective? I also added you as collaborator to this repository, so feel free to modify things on this branch. Also [optionally] you are welcome to add yourself to authors and copyright holders.

Thanks!

Best,
Robert

### CODE REVIEW

**Comment by haesleinhuepf** [comment]
The tests pass locally and that's sufficient btw.

**Comment by zoccoler** [comment]
Hi Robert,

The suggested re-arrangement makes sense to me. The main reason I had left the functions return just the data apart from the layer properties was because, if these functions were to be called by code (even without napari), they would return just the data, which makes sense. But I think any user doing that could just get the first element of the tuple.

I made a small change because the napari-plot-profile was duplicated when accessed from the plugins menu. Also, the negative part (as a 3D images and a surface) was not getting reversed. With these changes, this gets fixed, but only when the plugin is accessed from the Tools menu. Check the screens below:

From Tools Menu:
image

From Plugins Menu:
image

I could not find what is causing this. Do you have any guesses?

Best,
Marcelo

File: napari_plot_profile/_functions.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
+from typing import List
+
+import numpy as np
+from napari.types import LayerDataTuple, ImageData
+from skimage import measure
+from napari.utils.colormaps import colormap_utils
+
```
Comment (jo-mueller) [inline napari_plot_profile/_functions.py:7]:
```suggestion
import napari
```

**Comment by zoccoler** [comment]
Thanks Johannes @jo-mueller , I am converting this to draft to give it a try!

**Comment by zoccoler** [comment]
It is working now, thank you Johannes @jo-mueller !
Tests pass locally.
Robert @haesleinhuepf , could you please give it a look again?

**Comment by zoccoler** [comment]
Hi Robert @haesleinhuepf , 

do you think this can be merged? 😄

**Comment by haesleinhuepf** [comment]
Apologies for the delay @zoccoler . Good that you poked here! :-)

**Pull request merged**